### PR TITLE
TP-662 Add certificate update view

### PR DESCRIPTION
**Pull request by andyhd**
NOTE: Skipping descriptions update as @CJWaszczuk is working on a generic solution

image

This change adds an edit page for certificates.

The edit page template and confirm-update template are generic and located in `common/jinja2/common/`. It should be possible to reuse them for all components, only the form needs to be custom.
To enable this, the templates make use of `._meta.verbose_name` and `._meta.verbose_name_plural`. These attributes can be overridden in the model Meta class (see `commodities/models.py`).
The `verbose_name` is also used in the `TrackedModel.get_url()` method.

### CODE REVIEW

File: certificates/views.py [hunk 1]
```diff
@@ -44,4 +49,36 @@ class CertificatesList(TamatoListView):
 class CertificateDetail(TrackedModelDetailView):
     model = Certificate
     template_name = "certificates/detail.jinja"
-    queryset = Certificate.objects.latest_approved().select_related("certificate_type")
+
+    def get_queryset(self):
+        return Certificate.objects.approved_up_to_transaction(
+            WorkBasket.current(self.request).transactions.order_by("pk").last(),
```
Comment (NDevox) [inline certificates/views.py:55]: transactions have `updated_at` and `created_at` which may be better to sort by. Although not sure if that fits the usecase.

File: certificates/views.py [hunk 2]
```diff
@@ -44,4 +49,36 @@ class CertificatesList(TamatoListView):
...
+        ).select_related("certificate_type")
+
+
+class CertificateUpdate(TrackedModelDetailMixin, DraftUpdateView):
```
Comment (NDevox) [inline certificates/views.py:59]: Needs permissions required.

**Review by simonwo — APPROVED** [review]
LGTM.

I have a question about whether we should be having tests of this stuff, but let's take that offline.

File: certificates/jinja2/certificates/edit.jinja [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      No current workbasket
+    {%- endif %}
+  {%- endwith %}
+  </p>
+{% endblock %}
+
+{% block form %}
+  {% call django_form(action=object.get_url("edit")) %}
+    {{ crispy(form) }}
+  {% endcall %}
```
[the remark this answers is not in the document]
Comment (andyhd) [reply certificates/jinja2/certificates/edit.jinja:35]: I actually am about to move these templates into `common` :)

File: certificates/views.py (hunk 1 above)
Comment (andyhd) [reply certificates/views.py:55]: Thanks, I will try to make it generic 👍

**Comment by andyhd** [comment]
Tests are coming, sorry

File: certificates/views.py (hunk 1 above)
Comment (andyhd) [reply certificates/views.py]: I've put this into a mixin that is used by all Certificate views, but I haven't made it generic to all TrackedModels

**Pull request merged**